routing/index_graph_starter_joints.hpp

```diff
@@ -0,0 +1,130 @@
+#pragma once
+
+#include "routing/segment.hpp"
+#include "routing/index_graph_starter.hpp"
+
+#include "geometry/point2d.hpp"
+
+#include <algorithm>
+#include <map>
+#include <queue>
+#include <set>
+#include <utility>
```

Comment:
Why do we need `#include <utility>` and `#include <algorithm>` here? Should we move them to cpp?

Reply:
done